devito/passes/clusters/blocking.py

```diff
@@ -100,16 +130,37 @@ def callback(self, clusters, prefix):
                     PARALLEL_IF_PVT}.intersection(c.properties[d]):
                 return clusters
 
+            # Pointless if there's no data reuse
+            if not self._has_data_reuse(c):
+                return clusters
+
         # All good, `d` is actually TILABLE
         processed = attach_property(clusters, d, TILABLE)
 
         return processed
 
 
+class AnalyzeDeviceAwareBlocking(AnalyzeBlocking):
+
+    def __init__(self, options):
+        super().__init__(options)
+
+        self.gpu_fit = options['gpu-fit']
+
+    def _make_key_hook(self, cluster, level):
+        return (is_on_device(cluster.functions, self.gpu_fit),)
```

Comment:
can this is_on_device be moved in init as a attr since it's reused in both

Reply:
I'm not sure how and to what purpose...

Reply:
sorry missed the cluster arg, nevermind.